### ISSUE TITLE
RequireThis treats local variables as properties

### ISSUE THREAD

**Issue by artembilan** [checkstyle/checkstyle#3423]
`/var/tmp $ javac Test.java`

```
```

`/var/tmp $ cat Test.java`

``` java
public class Test {

    private final String s = "foo";

    public String getS() {
        String s = null;
        s = "bar"; // ??? Reference to instance variable 's' needs "this.". [RequireThis] !!!
        return s; 
    }

    public String getS(String s) {
        s = "bar"; // no violation
        return s;
    }

}
```

`/var/tmp $ cat config.xml`

``` xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
        "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">
<module name="Checker">
    <module name="TreeWalker">
        <module name="RequireThis">
            <property name="checkMethods" value="false" />
            <property name="validateOnlyOverlapping" value="false" />
        </module>
    </module>
</module>
```

`/var/tmp $ java -jar checkstyle-7.1-all.jar -c config.xml Test.java`

```
Starting audit...
[ERROR] d:\tmp\Test.java:7:17: Reference to instance variable 's' needs "this.". [RequireThis]
Audit done.
Checkstyle ends with 1 errors.
```

---

I really expect that local variable doesn't overlap to the property.
If I'd use property I'd specify `this.` for it.

That works correctly before version `6.17`.

---

Thanks
##

**Comment by romani** [comment]
@MEZk , please take a look.

**Comment by romani** [comment]
Looks like to do not respect scopes of variable visibility, need to be fixed.

**Comment by averza** [comment]
Hi,
It is still not fixed in the latest Checkstyle 7.5.1...

**Comment by MEZk** [comment]
@averza 
Yes. The issue is not closed, so it has not fixed yet.

**Comment by romani** [comment]
pre-assigned to @sagar-shah94.

**Comment by ps-sp** [comment]
I am on it too.

**Comment by romani** [comment]
@sagar-shah94 please choose another issue.

**Comment by ps-sp** [comment]
class InputAllowNonOverlappingLocalVars {
    String s3 = null;
    void foo() {
        try {
            if (true) {
                String s3 = "Hello, World!";
                if (true) {
                    s3 = new String(); // No violation ?
                }
                else {
                    s3 = new String(); // No violation ?
                }
            }
        }
        catch (Exception ex) {
        }

    }
}

This would be the acceptable right ?

**Comment by ps-sp** [comment]
Also, when variableDeclarationFrame would be BLOCK_FRAME and there's no overlapping then there can never be a violation ! Correct me if I am wrong. Thanks.

**Comment by romani** [comment]
>This would be the acceptable behavior right ?

No violations in this case as Check is enforcing "this." To appear. There is no way to but "this." In your example.

**Comment by romani** [comment]
fix is merged

**Comment by romani** [comment]
For some reason I failed to actually press merge from web.
Now PR is merged.
